### PR TITLE
chore: update dependency of ESLint to version 8.1.0 or later

### PR DESCRIPTION
**Pull request by darcyYe**
Update dependency of ESLint to version 8.1.0 or later

## Linear Issue Reference

LOG-396

Tested by `pnpm lint` in both ./packages/core and ./packages/react path with up-to-date @silverhand/configs

image

Also tested `pnpm test` in repo root.

image

### CODE REVIEW

File: packages/client/package.json [hunk 1]
```diff
@@ -47,4 +47,4 @@
     "extends": "@silverhand/eslint-config"
   },
   "prettier": "@silverhand/eslint-config/.prettierrc"
-}
+}
```
Comment (gao-sun) [inline packages/client/package.json:50]: please ensure an empty line in the end of each file

File: packages/client/src/index.test.ts [hunk 2]
```diff
@@ -110,9 +110,10 @@ describe('LogtoClient', () => {
 
     test('claims restored', async () => {
       const storage = new MemoryStorage();
+      const { idToken } = await generateIdToken();
```
Comment (simeng-li) [inline packages/client/src/index.test.ts:113]: nit: could rename the param when using destructor

```js
const  { idToken: id_token } = await generateIdToken();
...

storage.setItem(LOGTO_TOKEN_SET_CACHE_KEY, {
        ...fakeTokenResponse,
        id_token,
      });
```

**Review by simeng-li — APPROVED** [review]
overall LGTM

**Review by wangsijie — APPROVED** [review]
LGTM

**Pull request merged**